app/gui/src/model/module.rs

```diff
@@ -515,6 +524,18 @@ pub struct UploadingFile {
     pub error:          Option<String>,
 }
 
+#[allow(missing_docs)]
+#[allow(missing_copy_implementations)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct ImportMetadata {
+    #[serde(skip_serializing_if = "core::ops::Not::not")]
+    #[serde(default, deserialize_with = "enso_prelude::deserialize_or_default")]
+    pub is_temporary: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default, deserialize_with = "enso_prelude::deserialize_or_default")]
+    pub info:         Option<ImportInfo>,
```

Comment:
Why do we keep ImportInfo in metadata? The only field of ImportInfo should be deducible from the AST.

Reply:
This made it easier to use the existing API for removing imports, which require `ImportInfo`. It also makes debugging the metadata slightly nicer, as it adds the text of the import to the metadata. But it is a bit of overhead, so I’ve removed it in favour of a new API to remove imports by ID.